dust-lambda-extension/extensions/dust

```diff
@@ -0,0 +1,294 @@
+#!/usr/bin/env bash
+##
+## Copyright (c) 2025, Crash Override, Inc.
+##
+## This file is part of the Chalk project
+## (see https://crashoverride.com/docs/chalk)
+##
+
+set -euo pipefail
+
+# Enable debug mode only if DEBUG environment variable is set
+[[ ${DEBUG:-} == "true" ]] && set -x
+
+# Logger helpers with consistent formatting and timestamps
+log_with_level() {
+	local level="$1"
+	local color="$2"
+	local message="$3"
+	printf "\e[%sm[%s] [%s] %s\e[0m\n" "$color" "$(date -Iseconds)" "$level" "$message" >&2
+}
+
+LOG_DEBUG() { if [[ ${DEBUG:-} == "true" ]]; then log_with_level "DEBUG" "0;34" "$1"; else true; fi; }
+LOG_INFO() { log_with_level "INFO" "0;32" "$1"; }
+LOG_ERROR() { log_with_level "ERROR" "0;31" "$1"; }
+LOG_WARNING() { log_with_level "WARN" "0;33" "$1"; }
+
+# Configurables (set via environment variables)
+readonly EXTENSION_NAME="${EXTENSION_NAME:-dust}"
+readonly MAX_FAILURES=${MAX_FAILURES:-5}
+
+# Constants
+readonly CHALK_BIN_PATH="$TASK_PATH/chalk"
+readonly CHALK_MARK_PATH="$TASK_PATH/chalk.json"
+readonly TASK_PATH="/var/task"
+readonly TMP_CHALK_PATH="/tmp/chalk"
+
+# Validate critical environment variable
+if [[ -z ${AWS_LAMBDA_RUNTIME_API:-} ]]; then
+	LOG_ERROR "AWS_LAMBDA_RUNTIME_API environment variable is not set"
+	exit 1
```

Comment:
common pattern to have a util function like `fatal` which logs an error and then `exit 1`:

https://github.com/crashappsec/setup-chalk-action/blob/44e40964dfd031b5d6b77ce5015a418c673ff649/setup.sh#L46-L49